spine-ios/Sources/Spine/Spine.Generated+Extensions.swift

```diff
@@ -289,34 +300,63 @@ internal enum FileSource {
-                                return
-                            }
-                            guard let temp else {
-                                continuation.resume(throwing: "Could not download file.")
-                                return
-                            }
-                            do {
-                                continuation.resume(returning: try Data(contentsOf: temp, options: []))
-                            } catch {
+                let lock = NSRecursiveLock()
+                nonisolated(unsafe)
+                var isCancelled = false
+                nonisolated(unsafe)
+                var taskHolder:URLSessionDownloadTask? = nil
+                return try await withTaskCancellationHandler {
+                    try await withCheckedThrowingContinuation { continuation in
+                        let task = URLSession.shared.downloadTask(with: url) { temp, response, error in
+                            if let error {
                                 continuation.resume(throwing: error)
+                            } else {
+                                guard let httpResponse = response as? HTTPURLResponse, httpResponse.statusCode == 200 else {
+                                    continuation.resume(throwing: URLError(.badServerResponse))
+                                    return
+                                }
+                                guard let temp else {
+                                    continuation.resume(throwing: "Could not download file." as SpineError)
+                                    return
+                                }
+                                do {
+                                    continuation.resume(returning: try Data(contentsOf: temp, options: []))
+                                } catch {
+                                    continuation.resume(throwing: error)
+                                }
+                            }
+                        }
+                        task.resume()
+                        let shouldCancel = lock.withLock {
+                            if !isCancelled {
+                                taskHolder = task
                             }
+                            return isCancelled
+                        }
+                        if shouldCancel {
+                            task.cancel()
                         }
                     }
-                    task.resume()
+                } onCancel: {
+                    lock.withLock {
+                        isCancelled = true
+                        let value = taskHolder
+                        taskHolder = nil
+                        return value
+                    }?.cancel()
                 }
             }
         }
     }
 }
 
-extension String: Error {
+public struct SpineError: ExpressibleByStringInterpolation, Error, Hashable {
+    
+    public let message: String
+    
+    public init(stringLiteral value: String) {
```

Comment:
In regards to the comment above, this could be changed like below, in which case printing the error would, like with the error extension, preserving previous functionality.

```swift
struct TestError: Error, CustomStringConvertible {
    let description: String
    init(_ description: String) {
        self.description = description
    }
}
```